src/core/ext/client_channel/client_channel_factory.c

```diff
@@ -85,3 +90,120 @@ grpc_arg grpc_client_channel_factory_create_channel_arg(
   arg.value.pointer.vtable = &factory_arg_vtable;
   return arg;
 }
+
+struct grpc_channel_targets_info {
```

Comment:
@markdroth Are you ok with changing the code over to use slice_hash_table from #8842 once that's merged? If you feel strongly about it, I could use avl right away, but I feel that a hash table is better suited for this usecase for performance reasons (unsubstantiated claim though).